src/styles/blocks/calender.scss

```diff
@@ -0,0 +1,47 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $base-shift;
+
+  @each $day, $number in $weekdays {
+    
+    &--start-day-#{$day} :first-child {
+      margin-left: ($number * $cell-size) + ($base-shift * $number);
+    }
+  }
+
+  @for $index from 28 through 31 {
+
+    &--month-length-#{$index} {
+      .calendar__day:nth-child(n + #{$index + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    display: flex;
+    width: $cell-size;
+    height: $cell-size;
+    background-color: #eee;
+    border: 1px solid black;
+    box-sizing: border-box;
+    justify-content: center;
+    align-items: center;
+    transition: background-color 0.5s, transform 0.5s;
```

Comment:
Would be great to create variables for all the colors and transition time

Reply:
they are repeated only once so that's why i didn't create variables for them.